job-server/src/main/scala/spark/jobserver/WebApi.scala

```diff
@@ -561,9 +562,18 @@ class WebApi(system: ActorSystem,
    *    GET /healthz              - return OK or error message
    */
   def healthzRoutes: Route = pathPrefix("healthz") {
-    //no authentication required
     get { ctx =>
-      ctx.complete("OK")
+      try {
+          val healthCheckClass = Class.forName(config.getString("spark.jobserver.healthcheck"))
+          val instance = healthCheckClass.newInstance.asInstanceOf[HealthCheckService]
+          instance.healthCheckImpl(ctx, supervisor, jobInfoActor, daoActor)
+        }
+        catch {
+          case ex: Exception => {
+            logger.error("Exception in healthz", ex)
+            ctx.complete(500, errMap("Exception while invoking health check"))
```

Comment:
nit:
Instead of 500, you should use `StatusCodes......`.

Reply:
+1 on this as well. There is a function `logAndComplete`

Reply:
Addressed